src/eal.cpp

```diff
@@ -68,13 +67,54 @@ namespace eal {
   }
 
   // Postion Function Functions
-  vector<double> getPosition(vector<double> coeffs, double time) {
+  // vector<double> coeffs = [[cx_n, cx_n-1, cx_n-2, ... cx_0],
+  //                          [cy_n, cy_n-1, cy_n-2, ... cy_0],
+  //                          [cz_n, cz_n-1, cz_n-2, ... cz_0]]
+  // The equations evaluated by this function are:
+  //                x = cx_n * t^n + cx_n-1 * t^(n-1) + ... + cx_0
+  //                y = cy_n * t^n + cy_n-1 * t^(n-1) + ... + cy_0
+  //                z = cz_n * t^n + cz_n-1 * t^(n-1) + ... + cz_0
+  vector<double> getPosition(vector<vector<double>> coeffs, double time) {
+    if ( time < 0.0) {
+      throw invalid_argument("Invalid input time, must be non-negative.");
```

Comment:
Why enforce non-negative time?

Reply:
Naively, it didn't seem meaningful to have a negative time

Reply:
CSM uses negative times, because it does image time relative to the center of the image. We should support that usage.

Reply:
:+1: